lib/ohai/plugins/linux/network.rb

```diff
@@ -105,6 +105,10 @@ def check_routing_table(family, iface, default_route_table)
         route_entry = Mash.new(destination: route_dest,
                                family: family[:name])
         %w{via scope metric proto src}.each do |k|
+          if k == "via" && route_ending =~ /\bvia\sinet6\s+([^\s]+)/
+            route_entry[k] = $1
+            next
+          end
           # http://rubular.com/r/pwTNp65VFf
           route_entry[k] = $1 if route_ending =~ /\b#{k}\s+([^\s]+)/
         end
```

Comment:
There doesn't seem to be a reason this new code needs to be inside the loop.

```suggestion
        %w{via scope metric proto src}.each do |k|
          # http://rubular.com/r/pwTNp65VFf
          route_entry[k] = $1 if route_ending =~ /\b#{k}\s+([^\s]+)/
        end
        route_entry["via"] = $1 if route_ending =~ /\bvia\sinet6\s+([^\s]+)/          
```

Reply:
The other regex uses `\s+` for whitespace so the new regex probably should as well.